File: examples/ruby/fixed-array-write/ArrayWrite.runar.rb
# ArrayWrite -- Ruby port of
# examples/ts/fixed-array-write/ArrayWrite.runar.ts.

require 'runar'

class ArrayWrite < Runar::StatefulSmartContract
  prop :table, FixedArray[Bigint, 4], default: [0, 0, 0, 0]

  def initialize
    super()
  end

  runar_public i: Bigint
  def bump(i)
    @table[i] = @table[i] + 1
    assert true
  end
end

File: examples/python/fixed-array-write/ArrayWrite.runar.py
from runar import (
    StatefulSmartContract, Bigint, FixedArray,
    public, assert_,
)


class ArrayWrite(StatefulSmartContract):
    # Python port of examples/ts/fixed-array-write/ArrayWrite.runar.ts.

    table: FixedArray[Bigint, 4] = [0, 0, 0, 0]

    def __init__(self):
        super().__init__()

    @public
    def bump(self, i: Bigint):
        self.table[i] = self.table[i] + 1
        assert_(True)
